src/main/cpp/src/xxhash64.cu

```diff
@@ -296,27 +320,187 @@ class device_row_hasher {
 
   /**
    * @brief Computes the hash value of an element in the given column.
+   *
+   * When the column is non-nested, this is a simple wrapper around the element_hasher.
+   * When the column is nested, this uses a seed value to serially compute each
+   * nested element, with the output hash becoming the seed for the next value.
+   * This requires constructing a new hash functor for each nested element,
+   * using the new seed from the previous element's hash. The hash of a null
+   * element is the input seed (the previous element's hash).
    */
   class element_hasher_adapter {
    public:
-    template <typename T, CUDF_ENABLE_IF(cudf::column_device_view::has_element_accessor<T>())>
+    class element_hasher {
+     private:
+      Nullate const _check_nulls;
+      hash_value_type const _seed;
+      hash_value_type const _null_hash;
```

Comment:
_seed and _null_hash are always the same, so we can remove one.

Reply:
> _seed and _null_hash are always the same, so we can remove one.

This is to maintain consistency with the definition of `cudf::experimental::row::hash::element_hasher`.

Reply:
No need to keep consistent. Because it's a new class and we can remove one variable.